contributors/design-proposals/local-storage-overview.md

```diff
@@ -0,0 +1,604 @@
+  ```yaml
+  apiVersion: v1
+  kind: Pod
+  metadata:
+    name: foo
+  spec:
+    <snip>
+    nodeTolerations:
+      - key: node.alpha.kubernetes.io/notReady
+        operator: TolerationOpExists
+        tolerationSeconds: 600
+      - key: node.alpha.kubernetes.io/unreachable
+        operator: TolerationOpExists
+        tolerationSeconds: 1200
+    schedulingFailureTimeoutSeconds: 600
+  ```
+
+  A new PV taint will be introduced to handle unhealthy volumes.  The addon or another external entity can monitor the volumes and add a taint when it detects that it is unhealthy. 
+  ```yaml
+  apiVersion: v1
+  kind: PersistentVolumeClaim
+  metadata:
+    name: foo
+  spec:
+    <snip>
+    pvTolerations:
+      - key: storage.kubernetes.io/pvUnhealthy
+        operator: TolerationOpExists
+  ```
+7. Once Alice decides to delete the database, she destroys the StatefulSet, and then destroys the PVCs.  The PVs will then get deleted and cleaned up according to the reclaim policy, and the addon adds it back to the cluster.
+
+### Bob manages a distributed filesystem which needs access to all available storage on each node
+
+1. The cluster that Bob is using is provisioned with nodes that contain one or more secondary partitions
+2. The cluster administrator runs a DaemonSet addon that discovers secondary partitions across all nodes and creates corresponding PVs for them.
+3. The addon will monitor the health of secondary partitions and mark PVs as unhealthy whenever the backing local storage devices have failed.
+4. Bob creates a specialized controller (Operator) for his distributed filesystem and deploys it.
+5. The operator will identify all the nodes that it can schedule pods onto and discovers the PVs available on each of those nodes. The operator has a label selector that identifies the specific PVs that it can use (this helps preserve fast PVs for Databases for example).
+6. The operator will then create PVCs and manually bind to individual local PVs across all its nodes.
+7. It will then create pods, manually place them on specific nodes (similar to a DaemonSet) with high enough priority and have them use all the PVCs created by the Operator on those nodes.
+8. If a pod dies, it will get replaced with a new pod that uses the same set of PVCs that the old pod had used.
+9. If a PV gets marked as unhealthy, the Operator is expected to delete pods if they cannot tolerate device failures
+
+### Phippy manages a cluster and intends to mitigate storage I/O abuse
+
+1. Phippy creates a dedicated partition with a separate device for her system daemons. She achieves this by making `/var/log/containers`, `/var/lib/kubelet`, `/var/lib/docker` (with the docker runtime) all reside on a separate partition.
+2. Phippy is aware that pods can cause abuse to each other.
+3. Whenever a pod experiences I/O issues with it's EmptyDir volume, Phippy reconfigures those pods to use Persistent Volumes whose lifetime is tied to the pod.
```

Comment:
I meant to say that "use a persistent volume whose lifetime will be tied to that of the pod instead of an emptyDir volume". As of now PVs lifetime is outside that of a pod.